### ISSUE TITLE
PPM pipe

### ISSUE THREAD

**Issue by kowpa1990** [danielgtaylor/jpeg-archive#16]
hi,

I have a problem with compressing images from standard input using jpeg-recompress.exe. The source files are different, so I have to use a program that opens various image files, eg. ffmpeg. + I use an additional program [BATCHENC](http://members.home.nl/w.speek/batchenc.htm). The problem occurs when streaming ppm to jpeg-recompress.

example
```
ffmpeg.exe -i <infile> -f image2pipe -vcodec ppm - | jpeg-recompress.exe --ppm - out.jpg
```
there is an error
```
Incorrect image size! 374 vs. 5992721
av_interleaved_write_frame(): Broken pipe
frame=    1 fps=0.0 q=0.0 Lsize=    5852kB time=00:00:00.04 bitrate=1198544.2kbits/s
video:5852kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000000%
Conversion failed!
```

example 2
ImageMagick also does not work
```
convert.exe <infile> ppm:- | jpeg-recompress.exe --method smallfry --ppm - out.jpg
```
there is an error
```
Incorrect image size! 491 vs. 5992721
```

Am I doing something wrong?

**Comment by danielgtaylor** [comment]
This should work. I'll investigate and try to figure out what is going on.

**Comment by danielgtaylor** [comment]
@kowpa1990 this should now be fixed in the 2.1.1 release!

**Comment by IllyaMoskvin** [comment]
As an aside for any troubleshooters passing through, I got the `Incorrect image size!` error while processing some TIFF files. It turned out that several of the TIFFs were layered. When opened in Photoshop, they contained a single layer that hasn't been flattened into the background. Programmatically, they were thus interpreted as having *two* layers, with interesting implications.

Hitting them with ImageMagick's `convert in.tiff out.ppm` produced a PPM that reported its image size as precisely half of the image data it contained. Feeding that PPM into `jpeg-recompress` produced this error. Likewise, opening the TIFFs in Photoshop, flattening the image, and saving would halve their file size.

Interestingly, these TIFFs also seemed to also contain two identical color profiles, one for each layer, and hitting them with ImageMagick's `convert in.tiff out.icc` produced two `.icc` files (`out-0.icc` and `out-1.icc`).

I fixed the problem by adding a `-flatten` flag to my `convert` calls ([docs](http://www.imagemagick.org/Usage/layers/#flatten)). Here are my generalized convert and `jpeg-recompress` commands, for reference:

    convert "in.tif" -flatten "in.icc"
    convert "in.tif" -flatten -depth 8 "out.ppm"
    jpeg-recompress --quality high --accurate --method ssim --ppm "out.ppm" "out.jpg"
    convert "out.jpg" -density 150 -profile "in.icc" "out.jpg"
